src/librustc_mir/interpret/eval_context.rs

```diff
@@ -95,6 +134,52 @@ pub struct Frame<'mir, 'tcx: 'mir> {
     pub stmt: usize,
 }
 
+impl<'mir, 'tcx: 'mir> Eq for Frame<'mir, 'tcx> {}
+
+impl<'mir, 'tcx: 'mir> PartialEq for Frame<'mir, 'tcx> {
+    fn eq(&self, other: &Self) -> bool {
+        let Frame {
+            mir,
+            instance: _,
+            span: _,
+            return_to_block,
+            return_place,
+            locals,
+            block,
+            stmt,
+        } = self;
+
+        ptr::eq(mir, &other.mir)
+            && *return_to_block == other.return_to_block // TODO: Are these two necessary?
```

Comment:
Probably not in a global way (due to `stmt` on the upper stack frame), but still nice to have this impl be very correct.